services/provider/server/server.go

```diff
@@ -662,6 +629,58 @@ func storageClaimCephCsiSecretName(secretType, suffix string) string {
 
 // GetStorageClaim RPC call to get the ceph resources for the StorageClaim.
 func (s *OCSProviderServer) GetStorageClaimConfig(ctx context.Context, req *pb.StorageClaimConfigRequest) (*pb.StorageClaimConfigResponse, error) {
+
+	storageConsumer, err := s.consumerManager.Get(ctx, req.StorageConsumerUUID)
+	if err != nil {
+		return nil, err
+	}
+	storageCluster, err := util.GetStorageClusterInNamespace(ctx, s.client, s.namespace)
+	if err != nil {
+		return nil, err
+	}
+
+	consumerConfigMap := &v1.ConfigMap{}
+	if storageConsumer.Status.ResourceNameMappingConfigMap.Name == "" {
+		return nil, fmt.Errorf("waiting for ResourceNameMappingConfig to be generated")
+	}
+	consumerConfigMap.Name = storageConsumer.Status.ResourceNameMappingConfigMap.Name
+	consumerConfigMap.Namespace = storageConsumer.Namespace
+	err = s.client.Get(ctx, client.ObjectKeyFromObject(consumerConfigMap), consumerConfigMap)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get %s configMap. %v", consumerConfigMap.Name, err)
+	}
+	if consumerConfigMap.Data == nil {
+		return nil, fmt.Errorf("waiting StorageConsumer ResourceNameMappingConfig to be generated")
+	}
+
+	kubeResources, err := s.getKubeResources(ctx, storageConsumer)
+	if err != nil {
+		return nil, err
+	}
+
+	consumerConfig := util.WrapStorageConsumerResourceMap(consumerConfigMap.Data)
+
+	var extR []*pb.ExternalResource
+
+	if req.StorageConsumerUUID != "" {
+		if req.StorageClaimName == util.GenerateNameForCephBlockPoolStorageClass(storageCluster) {
+			extR, err = s.proxyGetStorageClaimConfigBlock(req, storageCluster, consumerConfig, kubeResources)
+			if err != nil {
+				klog.Errorf("failed to get storage claim config. %v", err)
+				return nil, status.Errorf(codes.Internal, "failed to get storage claim config. %v", err)
+			}
+		} else if req.StorageClaimName == util.GenerateNameForCephFilesystemStorageClass(storageCluster) {
```

Comment:
I am not sure you are comparing to the correct name. The `GenerateNameFor...` function takes rename input from the storage cluster while the old code didn't